### PR TITLE
Module6 branch

### PR DESCRIPTION
**Pull request by Linases**
image

Error is this part with 2 additional pixels

### CODE REVIEW

File: Browser_Actions/Browser_ActionTests.cs [hunk 1]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+            Assert.That(sortableDataTableUrl, Is.EqualTo($"{_mainUrl}/tables"));
+        }
+
+        public void SortByFirstName()
+        {
+            var firstNameHeader = _driver.FindElement(By.XPath("//*[@id='table1']//*[text()='First Name']"));
+            firstNameHeader.Click();
+            var allNamesList = _driver.FindElements(By.XPath("//*[@id='table1']//tbody/tr//td[2]"));
+            Assert.That(allNamesList.OrderBy(x => x.Text), Is.EqualTo(allNamesList), "'First Name' column is not sorted in ascending order");
+        }
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs:402]: usually private methods are located under teardown and teardown, below in the file

File: Browser_Actions/Browser_ActionTests.cs [hunk 2]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+
+        [Test]
+        public void RangeElement()
+        {
+            var elementHorizontalSlider = _driver.FindElement(By.XPath("//*[@id='content']//*[text()='Horizontal Slider']"));
+            elementHorizontalSlider.Click();
+            var elementHorizontalSliderUrl = _driver.Url;
+            Assert.That(elementHorizontalSliderUrl, Is.EqualTo($"{_mainUrl}/horizontal_slider"));
+            var inputRange = _driver.FindElement(By.XPath("//*[@type='range']"));
+            // inputRange.SendKeys(Keys.Home + "5") - this does not work somehow, input stays home value- "0"
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs]: great decision 👍👍 main - it works
but please remove comments))

File: Browser_Actions/Browser_ActionTests.cs [hunk 3]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+            var cellValue6 = cells[5].Text;
+
+            var expectedValue1 = "Smith";
+            var expectedValue2 = "John";
+            var expectedValue3 = "jsmith@gmail.com";
+            var expectedValue4 = "$50.00";
+            var expectedValue5 = "http://www.jsmith.com";
+            var expectedValue6 = "edit delete";
+
+            Assert.That(cellValue1, Is.EqualTo(expectedValue1));
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs]: maybe you can use cells[0].Text right is asserts? too many the same not clear names (cellValue1, cell2..)
maybe Assert.That(cells[0].Text, Is.EqualTo("Smith"));

File: Browser_Actions/Browser_ActionTests.cs [hunk 4]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+        [Test]
+        public void BasicAuth()
+        {
+            var elementBasicAuth = _driver.FindElement(By.XPath("//*[@id='content']//*[text()='Basic Auth']"));
+            elementBasicAuth.Click();
+            var BasicAuthUrl = _driver.Url;
+            Assert.That(BasicAuthUrl, Is.EqualTo($"{_mainUrl}/basic_auth"));
+
+            var login = "admin";
+            var authWindowLink = "https://" + login + ":" + login + "@" +
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs]: you can use just "admit : admin ...", separate variable makes code more difficult here

File: Browser_Actions/Browser_ActionTests.cs [hunk 5]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+            "the-internet.herokuapp.com/basic_auth";
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs:271]: I got it, strange task, not for alerts testing) will ask about how expected should it work

File: Browser_Actions/Browser_ActionTests.cs [hunk 6]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+
+            var validationMessage = _driver.FindElement(By.XPath("//*[@class='example']//p")).Text;
+            Console.WriteLine("Text is: " + validationMessage);
+            Assert.That(validationMessage, Is.EqualTo("Congratulations! You must have the proper credentials."));
+        }
+
+        [Test]
+        public void FileDownload()
+        {
+            var myDownloadFolder = @"C:\Users\lina.seskiene\SeleniumDownload";
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs]: That's absolute path, please use relative path

File: Browser_Actions/Browser_ActionTests.cs [hunk 7]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+            SortByFirstName();
+        }
+
+        [Test]
+        public void ToggleLastName()
+        {
+            OpenSortableTadaTables();
+            var lastNameHeader = _driver.FindElement(By.XPath("//*[@id='table1']//*[text()='Last Name']"));
+            var allLastNames = _driver.FindElement(By.XPath("//*[@id='table1']//tbody/tr//td[1]"));
+            lastNameHeader.Click();
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs]: there is double click in actions class

File: Browser_Actions/Browser_ActionTests.cs [hunk 8]
```diff
@@ -141,6 +144,268 @@ public void HeadlessMode()
...
+            OpenSortableTadaTables();
+            SortByFirstName();
+        }
+
+        [Test]
+        public void ToggleLastName()
+        {
+            OpenSortableTadaTables();
+            var lastNameHeader = _driver.FindElement(By.XPath("//*[@id='table1']//*[text()='Last Name']"));
+            var allLastNames = _driver.FindElement(By.XPath("//*[@id='table1']//tbody/tr//td[1]"));
```
Comment (Alina150601) [inline Browser_Actions/Browser_ActionTests.cs]: unused variable allLastNames

File: Browser_Actions/Browser_ActionTests.cs (hunk 1 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs:402]: Ok, moved private methods under teardown

File: Browser_Actions/Browser_ActionTests.cs (hunk 2 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs]: done

File: Browser_Actions/Browser_ActionTests.cs (hunk 3 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs]: true, it's less code. done

File: Browser_Actions/Browser_ActionTests.cs (hunk 4 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs]: ok, done

File: Browser_Actions/Browser_ActionTests.cs (hunk 6 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs]: Didn't succeeded transform from this absolute path to relative. So changed some code to create folder in existing directory during the test and get the path then.

File: Browser_Actions/Browser_ActionTests.cs (hunk 7 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs]: used, done.

File: Browser_Actions/Browser_ActionTests.cs (hunk 8 above)
Comment (Linases) [reply Browser_Actions/Browser_ActionTests.cs]: deleted.

**Pull request merged**